### PR TITLE
Eldrich recipes

### PR DESCRIPTION
**Pull request by EnderProyects**
Added recipes for eldrich maze blocks, to an existing research of gadomancy.
@Alastors

### CODE REVIEW

**Comment by Dream-Master** [comment]
@EnderProyects can you add a screen from the research ? Thanks

File: src/main/java/makeo/gadomancy/common/registration/RegisteredRecipes.java [hunk 1]
```diff
@@ -227,6 +234,112 @@ public static void init() {
...
+                    "CSC",
+                    "GCG",
+                    'C',
+                    new ItemStack(Blocks.mossy_cobblestone),
+                    'S',
+                    new ItemStack(ConfigBlocks.blockCrystal, 1, 6),
+                    'G',
+                    new ItemStack(Items.gunpowder));
+
+            // Glyphed Stone
```
Comment (Alastors) [inline src/main/java/makeo/gadomancy/common/registration/RegisteredRecipes.java:274]: No, this doesn't need to exist

**Review by Alastors — DISMISSED** [review]

File: src/main/java/makeo/gadomancy/common/registration/RegisteredRecipes.java (hunk 1 above)
Comment (EnderProyects) [reply src/main/java/makeo/gadomancy/common/registration/RegisteredRecipes.java:274]: ?

Comment (Alastors) [reply src/main/java/makeo/gadomancy/common/registration/RegisteredRecipes.java:274]: This is the one that shocks people isn't it

Comment (EnderProyects) [reply src/main/java/makeo/gadomancy/common/registration/RegisteredRecipes.java:274]: No lmao

Comment (EnderProyects) [reply src/main/java/makeo/gadomancy/common/registration/RegisteredRecipes.java:274]: Thats the non glowing one

Comment (EnderProyects) [reply src/main/java/makeo/gadomancy/common/registration/RegisteredRecipes.java:274]: The one that shocks is the one that doesnt has texture xd

**Comment by Alastors** [comment]
Show us the recipes

**Comment by EnderProyects** [comment]
image
image
image

**Comment by Dream-Master** [comment]
@Alastors @EnderProyects please add this pr to core mod it make no sense to put it to gadomancy

**Comment by Dream-Master** [comment]
It’s pure thaumcraft and has nothing todo with gadomancy? Or I am wrong ?

**Comment by Alastors** [comment]
> @Alastors @EnderProyects please add this pr to core mod it make no sense to put it to gadomancy 

Nah @Dream-Master  this would be fine to be in gadomancy, it uses entirely thaumcraft based materials, gadomancy has the ancient rune research already

**Comment by Alastors** [comment]
> It’s pure thaumcraft and has nothing todo with gadomancy? Or I am wrong ?

You're wrong, Gadomancy adds ancient block recipes, so this does make sense to go there, this is an extension of an existing set in Gado

**Comment by EnderProyects** [comment]
Why it failed again :(

**Comment by Alastors** [comment]
> image
> 
> image
> 
> image
> 
> 

Make the crusted use salis, reference the GT recipe, otherwise this'll turn into a salis source

**Comment by Dream-Master** [comment]
> > It’s pure thaumcraft and has nothing todo with gadomancy? Or I am wrong ?
> 
> You're wrong, Gadomancy adds ancient block recipes, so this does make sense to go there, this is an extension of an existing set in Gado

Core mod add gt support. Gadomancy not

**Comment by Dream-Master** [comment]
And gtnh gadomancy recipes are in core mod as well 😎

**Comment by Alastors** [comment]
> > > It’s pure thaumcraft and has nothing todo with gadomancy? Or I am wrong ?
> 
> > 
> 
> > You're wrong, Gadomancy adds ancient block recipes, so this does make sense to go there, this is an extension of an existing set in Gado
> 
> 
> 
> Core mod add gt support. Gadomancy not

We don't need GT support here

**Comment by EnderProyects** [comment]
> > image
> > image
> > image
> 
> Make the crusted use salis, reference the GT recipe, otherwise this'll turn into a salis source

At first it was going to use salis, but then I thouht that a mixed cristal cluster suits better, also the materials for the cristal cluster and the salis you can stract at the end  its like a 4 per 3, you lose pure shards if you go that way

**Comment by Alastors** [comment]
> And gtnh gadomancy recipes are in core mod as well 😎

Doesn't need Gtnh modification, the others in this set aren't modified by GTNH

**Comment by Dream-Master** [comment]
Have fun with fixing wci :trollface:

**Comment by Alastors** [comment]
> Have fun with fixing wci :trollface: 

wci?

**Comment by EnderProyects** [comment]
The wild card imports where already in there, how the other people that modified this code solved it? @Dream-Master

**Comment by Dream-Master** [comment]
Find it out if you want that code in this mod 🤷‍♂️

**Comment by EnderProyects** [comment]
> > > image
> > > image
> > > image
> > 
> > 
> > Make the crusted use salis, reference the GT recipe, otherwise this'll turn into a salis source
> 
> At first it was going to use salis, but then I thouht that a mixed cristal cluster suits better, also the materials for the cristal cluster and the salis you can stract at the end its like a 4 per 3, you lose pure shards if you go that way

I can just change the gundpowder for salis

**Comment by EnderProyects** [comment]
There are 51 wci in all the code, I could just solve all of them xd

**Comment by Alastors** [comment]
> There are 51 wci in all the code, I could just solve all of them xd

What does WCI mean

**Comment by Alastors** [comment]
I’m on mobile

**Comment by EnderProyects** [comment]
wci mean wildcard imports, but I didnt added any, they where already in the original code is the config.**; 
the (*)

**Comment by chochem** [comment]
yea we only recently started to enforce this.
your IDE can handle it automatically if you configure the settings.

**Comment by EnderProyects** [comment]
I´m in vscode, but I dont know how to configure it ( or where)

**Comment by Alastors** [comment]
> wci mean wildcard imports, but I didnt added any, they where already in the original code is the config.**; 
> 
> the (*) 
> 
> 

Thank you, I hate when people just say things and don't explain what they are.

Fixing those are whatever, just delete the import and re-import

**Comment by Alastors** [comment]
Aight as soon as you switch the glow stone (or whatever) to the appropriate amount of salis, I'll go ahead and approve.

**Comment by EnderProyects** [comment]
> Aight as soon as you switch the glow stone (or whatever) to the appropriate amount of salis, I'll go ahead and approve.

I already changed it

**Comment by Alastors** [comment]
Show new recipe

**Comment by EnderProyects** [comment]
image

**Comment by EnderProyects** [comment]
I thing that with just the mixed crystal cluster was enought, since one of this can turn the shards into 9 salis

**Comment by Alastors** [comment]
> image

I meant on the glowing crust lmfao

**Comment by EnderProyects** [comment]
I send this since the glowing one use  this one in the recipe

**Comment by EnderProyects** [comment]
image
The glowing one is this, but it uses the previous that uses 4 salis and 9 shards, and that 9 chards can be turned to salis, so is like 13 salis in 4 blocks, or 3.25 salis per block, and since you can stract only 3 salis per block if you are very lucky, you loose salis.
image

**Comment by EnderProyects** [comment]
I could just make those 4 glowstone to salis, but then is more expensive

**Comment by Rika-TH** [comment]
But the previous recipe makes 4 blocks, 1 of those blocks makes 12 of these, so 4 salis gives 12 crusteez adding those percentages you'd be getting more than 4 salis per cycle giving a net gain
(Not much, and very rng still, but definitely not a "loss at best")

**Comment by EnderProyects** [comment]
> But the previous recipe makes 4 blocks, 1 of those blocks makes 12 of these, so 4 salis gives 12 crusteez adding those percentages you'd be getting more than 4 salis per cycle giving a net gain (Not much, and very rng still, but definitely not a "loss at best")

You have to count the mixed shards in the midle, that can be turned into 9 salis

**Comment by Rika-TH** [comment]
> > But the previous recipe makes 4 blocks, 1 of those blocks makes 12 of these, so 4 salis gives 12 crusteez adding those percentages you'd be getting more than 4 salis per cycle giving a net gain (Not much, and very rng still, but definitely not a "loss at best")
> 
> You have to count the mixed shards in the midle, that can be turned into 9 salis

Just realised I missed em yeah, don't recall a mixed shard equalling 9 salis though? Isn't it like 2 salis?

**Comment by EnderProyects** [comment]
image
Each shard can be turned into 1 salis

**Comment by Rika-TH** [comment]
Ah, I think I see the angle now, the shards would require extra processing, but so does the crust so they're equal and you're just losing out on potential

I meant just for macerating straight into salis as that's how I would use this, unprocessed shards -> salis in a machine line as I'm too lazy to setup the essentials xP

Seeing that it's 3 balanced shards though really removes that idea anyways as you should only lose salis over time as that's 7 (direct) salis per set

**Comment by Rika-TH** [comment]
So even for someone trying to do some light cheese this'd be useless 👍

**Comment by EnderProyects** [comment]
Also that the research is post primordial pearl, at that time you can just dupe salis mundus
image

**Comment by Rika-TH** [comment]
you underestimate my stupidity xP had all that in LV, just don't like setting up essentia, furnaces transfer into alembics too slow for my liking, I like thaum, but being able to passive salis with minimal essentia work would actually be a decently big deal to me out of pure laziness

but after seeing the actual cluster now, that isn't a worry with this PR anyway, shouldn't have anything that statistically would be positive like that, only for building and such

**Comment by Rika-TH** [comment]
alators request seems to now be implemented correctly for balance, and cheaper for building from my view and your images :+1:

**Comment by Alastors** [comment]
> I could just make those 4 glowstone to salis, but then is more expensive

That was my original intention Lmfao

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (Dream-Master) [inline .vscode/settings.json]: this file not need to be in the pr.

Comment (Dream-Master) [reply .vscode/settings.json]: the file is still here @EnderProyects its not resolved

**Review by Alastors — APPROVED** [review]

**Pull request merged**